data/tools/wesnoth/wmltools3.py

```diff
@@ -286,66 +294,138 @@ def isresource(filename):
+            buffer.append(line[i])
+        elif line[i] == QUOTE:
+            open_token(QUOTE)
+            buffer.append(line[i])
+        elif line[i] == OPEN_BRACE:
+            open_token(OPEN_BRACE)
+            buffer.append(line[i])
+        elif line[i] == CLOSE_BRACE:
+            if wrapper_stack[-1] != OPEN_PARENS:
+                close_token(OPEN_BRACE)
+            if depth[OPEN_BRACE] == 0:
+                # Flush at end
+                if not prev_added_arg and len(buffer) > 0:
+                    added_arg = handle_argument(buffer)
                 break
             else:
-                arg += line[i]
-        elif line[i] == "(":
-            parendepth += 1
-        elif line[i] == ")":
-            parendepth -= 1
-        elif not line[i-1].isspace() and \
-             line[i].isspace() and \
-             brackdepth == 1 and \
-             parendepth == 0:
-            handle_argument()
-        elif not line[i].isspace() or parendepth > 0:
-            arg += line[i]
-    return (args, optional_args, brackdepth, parendepth)
+                buffer.append(line[i])
+        elif line[i] == OPEN_PARENS:
+            if wrapper_stack[-1] == OPEN_PARENS or wrapper_stack[-2:] == [OPEN_PARENS, EQUALS]:
+                # Char in an argument
+                buffer.append(line[i])
+            else:
+                if wrapper_stack[-1] == EQUALS or (wrapper_stack[-1] == OPEN_BRACE and \
+                    not line[i-1].isspace()):
+                    close_if_token(EQUALS)
+                    if depth[OPEN_BRACE] == 1 and not prev_added_arg:
+                        added_arg = handle_argument(buffer)
+                open_token(OPEN_PARENS)
+                if depth[OPEN_BRACE] != 1:
+                    buffer.append(line[i])
+        elif line[i] == CLOSE_PARENS:
+            # Source has too many closing parens.
+            quit = not close_token(OPEN_PARENS)
+            if depth[OPEN_BRACE] != 1:
+                buffer.append(line[i])
+            elif not prev_added_arg or line[i-1] == OPEN_PARENS:
+                # {MACRO arg1()} has two arguments
+                added_arg = handle_argument(buffer)
+            if quit:
+                added_arg = handle_argument(buffer)
+                break
+        elif line[i] == EQUALS and re.match(r'^([A-Z0-9_]+?)', line[i-1]):
+            open_token(EQUALS)
+            buffer.append(line[i])
+        elif line[i].isspace():
+            if line[i-1].isspace():
+                # Ignore consecutive spaces
+                continue
+            if not prev_added_arg and \
+             depth[OPEN_BRACE] == 1 and \
+             depth[OPEN_PARENS] == 0:
+                close_if_token(EQUALS)
+                added_arg = handle_argument(buffer)
+            buffer.append(line[i])
+        else:
+            buffer.append(line[i])
+
+        prev_added_arg = added_arg
+
+    args.pop(0)
+    return (args, optional_args, depth[OPEN_BRACE] > 0)
```

Comment:
We're missing the corresponding change to wmllint, and it doesn't seem to be in your open PRs / open branches. Please would you share it?

Reply:
Sorry for the trouble, @stevecotton . Patch in #7662.